### PR TITLE
Skip login on select reward

### PR DESCRIPTION
**Pull request by devton**

### CODE REVIEW

File: app/controllers/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -12,8 +12,7 @@ def self.add_providers
 
         flash[:notice] = I18n.t("devise.omniauth_callbacks.success", kind: p.name.capitalize)
         sign_in @auth.user, event: :authentication
-        redirect_to(session[:return_to] || root_path)
-        session[:return_to] = nil
+        redirect_to(session.delete(:fallback_redirect) || session.delete(:return_to) || root_path)
```
Comment (diogob) [inline app/controllers/omniauth_callbacks_controller.rb]: Should be
```ruby
redirect_to after_sign_in_path_for(@auth.user)
```

File: app/controllers/projects/contributions_controller.rb [hunk 2]
```diff
@@ -36,20 +36,22 @@ def show
...
 
-    # Select
     if params[:reward_id] && (@selected_reward = @project.rewards.find params[:reward_id]) && !@selected_reward.sold_out?
       @contribution.reward = @selected_reward
       @contribution.value = "%0.0f" % @selected_reward.minimum_value
     end
   end
 
+  def fallback_create
+    create
```
Comment (diogob) [inline app/controllers/projects/contributions_controller.rb]: Unnecessary, let's just use the create with a different route.

File: app/policies/contribution_policy.rb [hunk 3]
```diff
@@ -12,6 +12,14 @@ def resolve
     end
   end
 
+  def new?
+    record.project.online?
```
Comment (diogob) [inline app/policies/contribution_policy.rb]: Nice one ;)

File: app/controllers/projects/contributions_controller.rb [hunk 4]
```diff
@@ -57,6 +59,7 @@ def create
     @contribution.value = permitted_params[:contribution][:value]
     @contribution.referal_link = permitted_params[:contribution][:referal_link]
     @contribution.reward_id = (params[:contribution][:reward_id].to_i == 0 ? nil : params[:contribution][:reward_id])
+    session[:fallback_redirect] = fallback_create_project_contributions_path(params) unless current_user.present?
```
Comment (diogob) [inline app/controllers/projects/contributions_controller.rb]: Unnecessary, if we have a GET route for create, the default redirect should work.

Comment (devton) [inline app/controllers/projects/contributions_controller.rb]: nop, he get the POST request /contributions

File: app/policies/contribution_policy.rb [hunk 5]
```diff
@@ -12,6 +12,14 @@ def resolve
...
+  end
+
+  def fallback_create?
```
Comment (diogob) [inline app/policies/contribution_policy.rb]: We can remove this right?

**Pull request merged**